### PR TITLE
[ch11999] - Backoff library may cause panic on 32bit linux devices

### PR DESCRIPTION
**Pull request by rocos-jeremy-scott**
[ch11999]

### CODE REVIEW

File: backoff_test.go [hunk 1]
```diff
@@ -141,6 +141,23 @@ func TestConcurrent(t *testing.T) {
 	wg.Wait()
 }
 
+// func TestAtomicOperation_With32BitAttempt(t *testing.T) {
```
Comment (mohsenpashna) [inline backoff_test.go:144]: Can you please clarify this block of the commented test?

File: definition_32bit.go [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+package backoff
+
+import (
+	"math"
+	"sync/atomic"
+    "time"
+)
+
+const maxDuration = float64(math.MaxInt32 - 512)
```
Comment (mohsenpashna) [inline definition_32bit.go:11]: Would that be okay to use `float64` in 32 architecture? I assumed you tested with a 32bit device?

**Review by mohsenpashna — COMMENTED** [review]
Just a couple of questions/comments.

File: backoff_test.go (hunk 1 above)
Comment (mohsenpashna) [reply backoff_test.go:144]: Maybe just add a comment line?

File: definition_32bit.go (hunk 2 above)
Comment (rocos-jeremy-scott) [reply definition_32bit.go:11]: Yes, using 64 bit values is fine on 32 bit hardware. This code was running fine on a 32 bit raspberry pi.

File: backoff_test.go (hunk 1 above)
Comment (rocos-jeremy-scott) [reply backoff_test.go:144]: I'll remove the test as I'm not sure it should go in this library. The test also won't be run in a way that verifies everything is working correctly as that would require the test to be run on 32 bit hardware.

**Review by mohsenpashna — APPROVED** [review]
LGTM

**Review by edwinhayes — APPROVED** [review]
LGTM.

**Pull request merged**